src/uct/tcp/tcp_listener.c

```diff
@@ -15,62 +16,70 @@ static void uct_tcp_listener_conn_req_handler(int fd, void *arg)
     char ip_port_str[UCS_SOCKADDR_STRING_LEN];
     struct sockaddr_storage client_addr;
     ucs_async_context_t *async_ctx;
-    uct_tcp_sa_arg_t *sa_arg_ctx;
+    uct_tcp_sockcm_ep_t *ep;
+    uct_ep_params_t params;
     ucs_status_t status;
     socklen_t addrlen;
-    int accept_fd;
+    int conn_fd;
 
     ucs_assert(fd == listener->listen_fd);
 
-    async_ctx = listener->super.cm->iface.worker->async;
-
     addrlen   = sizeof(struct sockaddr_storage);
     status    = ucs_socket_accept(listener->listen_fd,
                                   (struct sockaddr*)&client_addr,
-                                  &addrlen, &accept_fd);
+                                  &addrlen, &conn_fd);
     if (status != UCS_OK) {
         return;
     }
 
-    ucs_assert(accept_fd != -1);
+    ucs_assert(conn_fd != -1);
 
-    ucs_trace("server accepted a connection request from client %s",
-              ucs_sockaddr_str((struct sockaddr*)&client_addr, ip_port_str,
-                               UCS_SOCKADDR_STRING_LEN));
+    ucs_trace("server accepted a connection request (fd=%d) from client %s",
+              conn_fd, ucs_sockaddr_str((struct sockaddr*)&client_addr,
+                                        ip_port_str, UCS_SOCKADDR_STRING_LEN));
 
     /* Set the accept_fd to non-blocking mode
      * (so that send/recv won't be blocking) */
-    status = ucs_sys_fcntl_modfl(accept_fd, O_NONBLOCK, 0);
+    status = ucs_sys_fcntl_modfl(conn_fd, O_NONBLOCK, 0);
     if (status != UCS_OK) {
         goto err;
     }
 
-    sa_arg_ctx = ucs_malloc(sizeof(uct_tcp_sa_arg_t), "accept sa_arg_ctx");
-    if (sa_arg_ctx == NULL) {
-        ucs_error("failed to allocate memory for a listen_ctx");
+    params.field_mask        = UCT_EP_PARAM_FIELD_CM               |
+                               UCT_EP_PARAM_FIELD_CONN_REQUEST     |
+                               UCT_EP_PARAM_FIELD_SOCKADDR_CB_FLAGS;
+    params.cm                = listener->super.cm;
+    params.conn_request      = NULL;
```

Comment:
why NULL? should not request be available inside `uct_tcp_listener_conn_req_handler`?

Reply:
the actual conn_request that will be used is the one that will be provided by the user when calling uct_ep_create, after invoking the listener's conn_req callback. the one here won't be used anyway,

Reply:
maybe then remove `UCT_EP_PARAM_FIELD_CONN_REQUEST` at all?

Reply:
it should be present though when creating an ep from an api call...
here, it's kind of an initial partial creation of a tcp sockcm ep for the server side.

Reply:
if you need it to detect client/server side inside ep creation, you can do it checking sockaddr presence in params, client must have it in its params. passing NULL conn_request from conn_request handler seems wired.

Reply:
it's already checked with UCT_EP_PARAM_FIELD_SOCKADDR.
when calling ep_create from an api, we do want the server to pass this flag so we should check for it.
any value i set in this case, won't be used anyway since this is not an api call...

Reply:
ok